### DISCUSSION TITLE
Duplicate fields

### DISCUSSION THREAD

**Discussion by javacode123** [prisma/prisma#22607] [Q&A]
i have two table
```
CREATE TABLE `level` (
  `id` int unsigned NOT NULL AUTO_INCREMENT COMMENT '等级ID',
  `upgrade_conditions ` varchar(255) DEFAULT NULL COMMENT '降级条件',
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci ROW_FORMAT=DYNAMIC COMMENT='分销商等级表';

CREATE TABLE `upgrade_conditions` (
  `id` int unsigned NOT NULL AUTO_INCREMENT COMMENT '自动升级条件ID',
  PRIMARY KEY (`id`),
  KEY `upgrade_conditions_ibfk_1` (`level_id`),
  CONSTRAINT `upgrade_conditions_ibfk_1` FOREIGN KEY (`level_id`) REFERENCES `level` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci ROW_FORMAT=DYNAMIC COMMENT='升级条件表';
```
upgrade_conditions has constraint with level.id, but when introspect prisma , it get duplicate field
![image](https://github.com/prisma/prisma-engines/assets/11720169/21465698-b64b-43cf-94ec-8ec16188317e)

**Comment by janpio** [comment]
This is probably expected, as you really have a field called `upgrade_conditions`, and also a foreign key in the other table `upgrade_condictions` that points to this one - where the automatically generated name is the same `upgrade_condictions`, hence the clash.

What would you expect instead?

**Reply by javacode123** [reply]
for this scene, maybe we can add a filed named {table_name}_ref

**Reply by janpio** [reply]
That would be worse than just using `{table_name}` in 99.999% of the cases though, right?
Also `_ref` is an arbitrary choice, that I am not even sure what it stands for. There are not terms in Prisma context that start with `ref` beside the `reference` - which is something different.

**Reply by javacode123** [reply]
yes. Any reasonable storage design should not include a filed with the same name as a table where the foreign key in. I think the first question to discuss is whether prisma cli should handle this situation. If the answer is yes. We should give the filed a unique name. if the answer is no 「I also do not have good idea about naming this field」, we just need return err.

**Reply by janpio** [reply]
> if the answer is no 「I also do not have good idea about naming this field」, we just need return err.

That is exactly what we have decided, and the error message should be pretty clear in this case that one of the two fields needs to be renamed.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
